File: testdata/bench/swift/BenchmarkTest00114.swift
import Vapor

func handler114(_ req: Request) throws -> Response {
    let page = try req.parameters.require("page")
    guard !page.contains("://") && !page.contains("..") else { throw Abort(.forbidden) }
    return req.redirect(to: "https://myapp.com/" + page)
}

File: testdata/bench/kotlin/BenchmarkTest00114.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*

// Safe: allowlist of redirect paths
suspend fun handler00114(call: ApplicationCall) {
    val page = call.parameters["page"]
    val allowed = mapOf("home" to "/", "profile" to "/profile", "settings" to "/settings")
    val target = allowed[page] ?: "/"
    call.respondRedirect(target)
}
